src/realm/group.cpp

```diff
@@ -738,18 +738,23 @@ TableRef Group::add_table_with_primary_key(StringData name, DataType pk_type, St
         throw LogicError(LogicError::detached_accessor);
     check_table_name_uniqueness(name);
 
-    if (Replication* repl = *get_repl())
-        repl->add_class_with_primary_key(name, pk_type, pk_name, nullable);
-
-    auto table = do_add_table(name);
-
-    ColKey pk_col = table->add_column(pk_type, pk_name, nullable);
+    auto table = do_add_table(name, false, false);
+
+    // Add pk column - without replication
+    ColumnAttrMask attr;
+    if (nullable)
+        attr.set(col_attr_Nullable);
+    ColKey pk_col = table->generate_col_key(ColumnType(pk_type), attr);
+    table->do_insert_root_column(pk_col, ColumnType(pk_type), pk_name);
     table->do_set_primary_key_column(pk_col);
 
+    if (Replication* repl = *get_repl())
+        repl->add_class_with_primary_key(table->get_key(), name, pk_type, pk_name, nullable);
+
```

Comment:
There is something fishy here - also in the original code. In line 754 below, if it is needed to check for table being null, then dereferencing it without a check in line 747 and forward may crash. I think we should be able to remove the check.

Reply:
Sure